docs/library/graphing/other-charts/pyplot.md

```diff
@@ -0,0 +1,142 @@
+
+
+## Installation
+
+Install the `reflex-pyplot` package using pip.
+
+```bash
+pip install reflex-pyplot
+```
+
+## Basic Example
+
+To display a Matplotlib plot in your app, you can use the `pyplot` component. Pass in the figure you created with Matplotlib to the `pyplot` component as a child.
+
+```md alert info
+# You must close the figure after creating 
+
+By not closing the figure, it could cause memory issues.
+```
+
+```python demo exec
+import matplotlib.pyplot as plt
+import reflex as rx
+import numpy as np
+
+def create_contour_plot():
+    X, Y = np.meshgrid(np.linspace(-3, 3, 256), np.linspace(-3, 3, 256))
+    Z = (1 - X/2 + X**5 + Y**3) * np.exp(-X**2 - Y**2)
+    levels = np.linspace(Z.min(), Z.max(), 7)
+    
+    fig, ax = plt.subplots()
+    ax.contourf(X, Y, Z, levels=levels)
+    plt.close(fig)
+    return fig
+
+def pyplot_simple_example():
+    return rx.card(
+            pyplot(create_contour_plot(), width="100%", height="400px"),
+            bg_color='#ffffff',
+            width="100%",
+        )
+```
+
+## Stateful Example
+
+Lets create a scatter plot of random data. We'll also allow the user to randomize the data and change the number of points. 
+
+In this example, we'll use a `color_mode_cond` to display the plot in both light and dark mode. We need to do this manually here because the colors are determined by the matplotlib chart and not the theme.
+
+```python demo exec
+def create_plot(theme: str, plot_data: tuple, scale: list):
+    bg_color, text_color = ('#1e1e1e', 'white') if theme == 'dark' else ('white', 'black')
+    grid_color = '#555555' if theme == 'dark' else '#cccccc'
+    
+    fig, ax = plt.subplots(facecolor=bg_color)
+    ax.set_facecolor(bg_color)
+    
+    for (x, y), color in zip(plot_data, ["#4e79a7", "#f28e2b", "#59a14f"]):
+        ax.scatter(x, y, c=color, s=scale, label=color, alpha=0.6, edgecolors="none")
+    
+    ax.legend(facecolor=bg_color, edgecolor='none', labelcolor=text_color)
+    ax.grid(True, color=grid_color)
+    ax.tick_params(colors=text_color)
+    for spine in ax.spines.values():
+        spine.set_edgecolor(text_color)
+    
+    for item in [ax.xaxis.label, ax.yaxis.label, ax.title]:
+        item.set_color(text_color)
+    
+    return fig
+
+class PyplotState(rx.State):
+    num_points: int = 100
+    plot_data: tuple = tuple(np.random.rand(2, 100) for _ in range(3))
+    scale: list = [random.uniform(0, 100) for _ in range(100)]
+    
+    def randomize(self):
+        self.plot_data = tuple(np.random.rand(2, self.num_points) for _ in range(3))
+        self.scale = [random.uniform(0, 100) for _ in range(self.num_points)]
+    
+    def set_num_points(self, num_points: list[int]):
+        self.num_points = num_points[0]
+        self.randomize()
+    
+    @rx.var
+    def fig_light(self) -> plt.Figure:
+        fig = create_plot('light', self.plot_data, self.scale)
+        plt.close(fig)  # Close the figure after creating it
```

Comment:
Can this go within the `create_plot` code so they don't have to call it outside each time?